sdk/java-sdk/src/main/scala/kalix/javasdk/impl/GrpcClients.scala

```diff
@@ -104,13 +103,14 @@ final class GrpcClients(system: ExtendedActorSystem) extends Extension {
   def getGrpcClient[T](serviceClass: Class[T], service: String): T =
     getGrpcClient(serviceClass, service, port = 80, remoteAddHeader)
 
-  /** Local gRPC clients point to services (user components or Kalix services) in the same deployable */
+  /** gRPC clients point to services (user components or Kalix services) in the same deployable */
   private def getLocalGrpcClient[T](serviceClass: Class[T]): T = {
-    proxyHostname match {
-      case Some("localhost") => getGrpcClient(serviceClass, "localhost", selfPort.getOrElse(9000), localAddHeader)
-      case Some(selfName)    => getGrpcClient(serviceClass, selfName, 80, localAddHeader)
-      case None =>
-        throw new IllegalStateException("Self service name not set by proxy at discovery, too old proxy version?")
+    (proxyHostname, proxyPort) match {
+      case (Some(hostname), Some(port)) =>
+        getGrpcClient(serviceClass, hostname, port, localAddHeader)
+      case _ =>
+        throw new IllegalStateException(
+          "Service proxy hostname and port are not set by proxy at discovery, too old proxy version?")
```

Comment:
As discussed on the stand-up today, we still need to support the old code because a user can update their SDK version, but still have the docker-compose using an old version of the proxy. 

When that happens, users will probably revert their SDK version believing that the SDK is broken. It will only broken in dev mode though. 

So, we need to add two cases to this pattern matching.

```suggestion
    (proxyHostname, proxyPort) match {
      case (Some(hostname), Some(port)) =>
        getGrpcClient(serviceClass, hostname, port, localAddHeader)
      case (Some("localhost"), None) => getGrpcClient(serviceClass, "localhost", selfPort.getOrElse(9000), localAddHeader)
      case (Some(selfName), None)    => getGrpcClient(serviceClass, selfName, 80, localAddHeader)
      case _ =>
        throw new IllegalStateException(
          "Service proxy hostname and port are not set by proxy at discovery, too old proxy version?")
```

Basically, if port is not set (old proxy behavior), we fallback to the old strategy.

Same for the javascript sdk